addons/mail/static/src/core/common/message.js

```diff
@@ -459,6 +479,26 @@ export class Message extends Component {
         }
     }
 
+    async redirectMessage(messageId) {
+        const { threadData, error } = await rpc("/mail/message/redirect", {
+            message_id: messageId,
+        });
+        if (error) {
+            const errorMessage =
+                error === "Unauthorized"
+                    ? _t("You do not have access to this conversation.")
+                    : error === "NotFound"
+                    ? _t("This message no longer exists.")
+                    : _t("Something went wrong.");
+            this.notification.add(errorMessage, { type: "danger" });
+            return;
+        }
+        const [thread] = this.store.insert(threadData).Thread;
+        if (thread.model === "discuss.channel") {
+            thread.open();
+        }
```

Comment:
Why only limiting to just `discuss.channel`?
I'd expect the feature to work fine for all message links, including from chatter.
If we don't want this feature enabled in chatter, I'd expect to just not enabling the link but if we had link prior then this should still work regardless of thread at hand.

Also context can be deduce from `redirectMessage` (when you move in `open()` method of `Message` model like a suggested in comment before). It's not a good idea to destructure from data as this leaks how data are formatted, which you shouldn't necessarily know... And this is even weak in case we retrieve more data, e.g. what if formatter returns 2 threads instead of 1? We plan to have sub-thread feature soon, so nothing really enforces there should be only 1 thread data in the formatter!

Reply:
The reason for limiting it to `discuss.channel` is that only discuss channels should be opened in `Discuss/Chatwindow`.
For other threads, if the message being redirected is present in the same chatter it would be highlighted there itself, but if it is of other thread than the record would need to open in new window. So, `thread.open()` only makes sense for `discuss.channel` threads.

I get your point being error prone code. I would do something about it.

Reply:
Opening a chatter in chat window is an actual feature that already exists. It's not well known because you get it when you set Notification Preferences to `Handle in Odoo` (non-default value) and you click on a messaging menu notification item from chatter.

So there's no need to limit to `discuss.channel`. `Thread.open()` is designed to work for all kinds of threads :)

Reply:
I know about the chatter able to open in chatwindow. But, the spec is that the chatter should be opened in new window, with it's record.
Opening a message with it's record is more better rather than just message in a chatwindow.

Reply:
I'd expect exact same place for code to redirect to message, whether the message comes from `discuss.channel` or chatter.

To follow the slightly different functional specs while keeping code as similar as possible, I would put the handling of chatter message opening here too.

Reply:
I see that the guard is there to prevent the side-effect to open the chatter message in the current tab, as you made the opening of new tab in the route of chatter itself.

Honestly this is quite confusing, and this LOC shows that this is not ok. You have to be aware of the implementation in Python to understand why we silently ignore chatter message redirection. This looks decent enough in this PR because we see the diff, but the likelihood to change one part of code and not the other is high.

Locality of concern is an important aspect in keeping the code maintainable. I would improve it, e.g. by handling chatter messages here too. This could be `thread.open({ newTab: true })`. Again this effectively gives the same behaviour as you implemented, but at least this is far more maintainable

Reply:
I get your point, but with doing this the issue would be that I would have to do 2 RPC calls for redirections.
1) On click which will get the message data `/mail/message/id`
2) Get request done by `ir.actions.act_url` action in new tab.

So, in case when the chatter should have been opened in new tab I would have to do both these calls where as it could have been done by a second one itself. The first one would be obsolete as I would have to all the checks in the later calls aswell.